outputs.tf

```diff
@@ -14,12 +14,6 @@ output "port" {
 }
 
 output "host" {
-  value = coalesce(
-    module.dns.hostname,
-    join(
-      "",
-      aws_elasticache_replication_group.default.*.primary_endpoint_address
-    )
-  )
+  value       = concat(aws_elasticache_replication_group.default.*.primary_endpoint_address, [""])[0]
```

Comment:
value       = module.dns.hostname != "" ? module.dns.hostname : join("", aws_elasticache_replication_group.default.*.primary_endpoint_address)

We need to check if `module.dns.hostname` exists and use it; otherwise use the primary endpoint address

Reply:
Sure. Makes sense that you'd be interested in one or the other (AWS URL or DNS).
RDS module outputs both the real endpoint and the DNS hostname. 
https://github.com/cloudposse/terraform-aws-rds-cluster/blob/master/outputs.tf#L21

Reply:
good point @kskewes, we can add it to the elasticache module as well.
we'll approve fast if you do it, thanks

Reply:
Please see latest commit. I see in the RDS module that there's no logic around `module.dns.hostname` so have stripped that back out here.